### PR TITLE
Sharing : Add wp_sharing_email_send_post_subject filter

### LINKED ISSUE #8165: Sharing: Add a filter to allow modifying the email share Subject Line
Originally suggested here:

https://wordpress.org/support/topic/edit-subject-line-in-sharing-emails/

We don't have a way of filtering the subject line; it could be useful to avoid cases such as these where the host is blocking such emails based on the subject line (likely because of #448 ).

https://github.com/Automattic/jetpack/blob/4ba5207a2d833238b342d523e758bd90efb45cb9/modules/sharedaddy/sharedaddy.php#L54

### PR DESCRIPTION
**Pull request by Umangvaghela**
Fixes #8165

#### Changes proposed in this Pull Request:

* add wp_sharing_email_send_post_subject  for allow modifying the email share Subject Line

#### Testing instructions:

*

### CODE REVIEW

File: modules/sharedaddy/sharedaddy.php [hunk 1]
```diff
@@ -51,7 +51,9 @@ function sharing_email_send_post( $data ) {
 	// Make sure to pass the title through the normal sharing filters.
 	$title = $data['sharing_source']->get_share_title( $data['post']->ID );
 
-	wp_mail( $data['target'], '[' . __( 'Shared Post', 'jetpack' ) . '] ' . $title, $content, $headers );
+	$subject = apply_filters('wp_sharing_email_send_post_subject',sprintf(__('Shared Post','jetpack')));
```
Comment (jeherve) [inline modules/sharedaddy/sharedaddy.php]: Could you add a docblock before the filter, as with the other filters in this file?

Comment (jeherve) [inline modules/sharedaddy/sharedaddy.php]: Could you add spacing around quotes, parenthesis, and before `sprintf`?

**Comment by Umangvaghela** [comment]
@jeherve 

Greetings of the day !!

Thanks for reply.
I add changes which is you suggested me. Please check it and provide the feedback for the same.

Thanks

File: modules/sharedaddy/sharedaddy.php [hunk 2]
```diff
@@ -51,7 +51,16 @@ function sharing_email_send_post( $data ) {
...
 
-	wp_mail( $data['target'], '[' . __( 'Shared Post', 'jetpack' ) . '] ' . $title, $content, $headers );
+	/**
+	 * Filter the Sharing Email Send Post Subject.
+	 *
+	 * @module sharedaddy
+	 *
+	 * @param string $var Sharing Email Send Post Subject. Default is "Shared Post".
+	 */
+	$subject = apply_filters( 'wp_sharing_email_send_post_subject', __( 'Shared Post', 'jetpack' ) );
```
Comment (jeherve) [inline modules/sharedaddy/sharedaddy.php]: Could you add square brackets (`[]`) around the subject line, like in the original message?

Thanks!

**Comment by Umangvaghela** [comment]
@jeherve 

Changes Done.

Thanks

File: modules/sharedaddy/sharedaddy.php [hunk 3]
```diff
@@ -51,7 +51,16 @@ function sharing_email_send_post( $data ) {
 	// Make sure to pass the title through the normal sharing filters.
 	$title = $data['sharing_source']->get_share_title( $data['post']->ID );
 
-	wp_mail( $data['target'], '[' . __( 'Shared Post', 'jetpack' ) . '] ' . $title, $content, $headers );
+	/**
+	 * Filter the Sharing Email Send Post Subject.
+	 *
+	 * @module sharedaddy
+	 *
```
Comment (jeherve) [inline modules/sharedaddy/sharedaddy.php]: Could you add `@since 5.7.0` here too?

**Comment by Umangvaghela** [comment]
@jeherve 

Done.

Thanks

File: modules/sharedaddy/sharedaddy.php [hunk 4]
```diff
@@ -51,7 +51,18 @@ function sharing_email_send_post( $data ) {
...
+	 * @since 5.7.0
+	 *
+	 * @param string $var Sharing Email Send Post Subject. Default is "Shared Post".
+	 */
+	$subject = apply_filters( 'wp_sharing_email_send_post_subject', '[' . __( 'Shared Post', 'jetpack' ) . '] ' );
```
Comment (oskosk) [inline modules/sharedaddy/sharedaddy.php]: I think something is missing here. 

The `$title` need to be concatenated in the subject.

File: modules/sharedaddy/sharedaddy.php [hunk 5]
```diff
@@ -51,7 +51,18 @@ function sharing_email_send_post( $data ) {
...
+
+	wp_mail( $data['target'], $subject, $title, $content, $headers );
```
Comment (oskosk) [inline modules/sharedaddy/sharedaddy.php]: Here `$title` shouldn't be the third argument. `$title should be concatenated into the subject`.

**Comment by Umangvaghela** [comment]
@oskosk 

Changes Done.

Thanks

**Comment by oskosk** [comment]
Thanks for the changes @Umangvaghela ! I've relabeled as needs review...

Code looks good now, but how do I test that the funcionality updated here is not affected? "Testing instructions" on the description of the PR, pleaseee 😄

**Comment by Umangvaghela** [comment]
@oskosk 

Sorry for late reply.

Testing instruction.

1 ) Enable sharing email option.
2 ) Add filter "wp_sharing_email_send_post_subject" in active theme function.php.
3 ) Update the subject if you want.
4 ) Share post by email sharing option.

Thanks

**Comment by Umangvaghela** [comment]
@oskosk 

Hello Sir 

Can you told me anything is missing for "testing description" or In PR. If you find anyhting else so told me I will change or add it as soon as possible. :) 


Thanks,

**Comment by Umangvaghela** [comment]
@oskosk

Please review changes because you give me changes and I add changes which is you suggested me as well as I was add testing instructions for test the functionality.

Thanks

**Comment by Umangvaghela** [comment]
@zinigor 

changes done.

Thanks

**Pull request merged**